src/common/reducers/snaps.js

```diff
@@ -0,0 +1,41 @@
+import * as ActionTypes from '../actions/snaps';
+import { parseGitHubRepoUrl } from '../helpers/github-url';
+
+export function snaps(state = {
+  isFetching: false,
+  success: false,
+  error: null,
+  snaps: null,
+}, action) {
+  switch(action.type) {
+    case ActionTypes.FETCH_SNAPS:
+      return {
+        ...state,
+        isFetching: true
+      };
+    case ActionTypes.FETCH_SNAPS_SUCCESS:
+      return {
+        ...state,
+        isFetching: false,
+        success: true,
+        snaps: action.payload.map((snap) => {
+          return {
+            // parse repository info to keep consistent data format
+            repo: parseGitHubRepoUrl(snap.git_repository_url),
```

Comment:
Maybe a bit redundant since `repo` can be derived trivially from `snap.git_repository_url`?  You could call `parseGitHubRepoUrl` from a `mapStateToProps` function.  Then `snaps` really would just contain snap objects.